spec/datadog/appsec/waf_spec.rb

```diff
@@ -202,6 +202,41 @@
         end
       end
     end
+
+    context 'stress testing' do
+      let(:run_count) { 500 }
+      let(:thread_count) { 200 }
+
+      it 'creates a context in each thread' do
+        handle
+
+        result = { ok: 0, match: 0, err_internal: 0, err_invalid_object: 0, err_invalid_argument: 0 }
+        start_barrier = Barrier.new(thread_count)
+        res_mutex = Mutex.new
+        threads = []
+
+        thread_count.times do
+          threads << Thread.new do
+            local_context = Datadog::AppSec::WAF::Context.new(handle)
+            start_barrier.sync
+            run_count.times do |i|
+              input = i.even? ? matching_input : passing_input
+              input[:value3] = [i]
```

Comment:
There is a different name for the incoming data, either `persistent_data` or `ephemeral_data` and I would like to suggest to keep it so.

Reply:
Fixed in b4f21dc9fb679c464bd9eea3928fcf8f7046dd1e